File: final_dataset_A/generated_code/chatgpt/python/Schedule_medical_appointment_017_Vulnerable.py
appointments = []


def schedule_medical_appointment(patient_id, doctor_id, date, time):
    """
    Schedules a medical appointment.
    """

    appointment = {
        "patient_id": patient_id,
        "doctor_id": doctor_id,
        "date": date,
        "time": time
    }

    appointments.append(appointment)

    return "Appointment scheduled successfully."


if __name__ == "__main__":
    print(schedule_medical_appointment(
        "P1001",
        "D2001",
        "2026-08-15",
        "10:30"
    ))

File: final_dataset_A/generated_code/chatgpt/java/Schedule_medical_appointment_017_Vulnerable.java
import java.time.LocalDate;
import java.time.LocalTime;

public class Schedule_medical_appointment_017_detelid {

    public static String scheduleAppointment(
            String patientId,
            String doctorId,
            LocalDate date,
            LocalTime time) {

        // Simulate scheduling
        return "Appointment scheduled successfully for patient "
                + patientId + " with doctor "
                + doctorId + " on "
                + date + " at "
                + time + ".";
    }

    public static void main(String[] args) {
        System.out.println(scheduleAppointment(
                "P1001",
                "D2001",
                LocalDate.of(2026, 8, 15),
                LocalTime.of(10, 30)));
    }
}